lib/repository.php

```diff
@@ -55,9 +55,11 @@ public static function add($entity, $save = true) {
 
     public static function edit($id, $entity) {
         if (array_key_exists($id, self::$entities)) {
-            self::$entities[$id] = array_merge(self::$entities[$id], $entity);
-            self::save();
-            return true;
+            if ((is_array(self::$entities[$id]) || is_object(self::$entities[$id])) && (is_array($entity) || is_object($entity))) {
+                self::$entities[$id] = array_merge(self::$entities[$id], $entity);
```

Comment:
Will array_merge work on objects? I think if it is an object, you should really replace the self::$entities[$id] one with the new object unless you are willing to do an object merge manually.

Reply:
I can convert Object  to Array. what do you think?

Reply:
The Hm_Contact object has properties and methods, converting to array will loose some functionality on the object. I prefer if you replace the entity. Something like:
```
if ($entity instanceof Hm_Contact) {
  self::$entities[$id] = $entity;
} else {
  self::$entities[$id] = array_merge(self::$entities[$id], $entity);
}
```

Reply:
I think I understand. Thanks